pkg/ebpf/compiler/shim.cpp

```diff
@@ -0,0 +1,47 @@
+#include "compiler.h"
+
+extern "C" {
+
+struct bpf_compiler {
+    ClangCompiler *cpp_compiler;
+};
+
+#include "shim.h"
+
+bpf_compiler *new_bpf_compiler(void)
+{
+    auto cpp_compiler = new ClangCompiler("clang");
+    auto c_compiler = new bpf_compiler();
+    c_compiler->cpp_compiler = cpp_compiler;
+    return c_compiler;
+}
+
+int bpf_compile_to_object_file(bpf_compiler *compiler, const char *input_file, const char *output_file, const char **cflagsv, char verbose)
+{
+    auto cppCompiler = (ClangCompiler*) compiler->cpp_compiler;
+    std::vector<const char*> cflags;
+    if (cflagsv) {
+        while(*cflagsv) {
```

Comment:
Could you get an empty string? Wondering if you need a check like
```c++
while(*cflagsv && **cflagsv != '\0')
```

Reply:
I don't think an empty string would cause problems.